homeassistant/components/risco/config_flow.py

```diff
@@ -51,6 +52,7 @@
         vol.Required(CONF_HOST): str,
         vol.Required(CONF_PORT, default=1000): int,
         vol.Required(CONF_PIN): str,
+        vol.Optional(CONF_COMMUNICATION_DELAY, default=0): int,
```

Comment:
If 1 second solves the problem for your device, why can't we set that for all devices, and not add an option? This option is not easy to set appropriately for an inexperienced user.

Reply:
We know that a lot (most?) of environments don't need any delay, and it's reasonable to suspect other environments might need longer delays (Risco communication is very finicky). I'm not sure we want to "punish" everyone based on the worst case scenario.

Reply:
(I do think it makes sense to always have the delay for the config flow though, but for the config entry setup I think the delay should be taken from the options)

Reply:
1 second isn't a very long delay in terms of user behavior or experience. This is a polling integration, right?

Is it possible to identify devices that we know need a delay? Could we do a retry in a loop with an increasing delay and see if that works and remember the delay that works?

Reply:
If this field belongs to an options flow it means that we either need to set a default for all panels, like 1 second or we do like Martin proposes and use an increasing delay until we are able to connect.
I would need to check if we can decrease the default timeout because I think it is 10 seconds at the moment so if we need to do 3 attempts it would result in a max 30 seconds delay to complete the config flow.
If I'm correct, this is a local push integration for local and cloud polling for cloud.

Reply:
The local variant is push, not poll. I'd hate to introduce a 1 second delay on every config entry load, just because of a minority of cases.
I'm not sure increasing delay would be a good option here because every try takes a while like @FredericMa mentioned.
My suggestion is:
1. 1 second delay for config flow (hoping we don't have to increase it in the future to cover more cases)
2. Option flow for ongoing delay

Reply:
I think we should explore all other options before adding this as an option.

Reply:
I've made some changes for testing yesterday to use an incremental delay during the setup of the integration. At the moment the limit is set to 3 seconds (= 3 attempts) but 2 or even 1 might already be enough to cover all cases. This delay is stored as an option and used for future use. It can also be changed in the options flow afterwards in case it needs to changed.

For my system it takes 28 seconds to setup with this PR and 33 seconds with the changes of yesterday (compare link with this PR at the bottom). The additional 5 second delay is needed to allow the panel to reset its encryption before we can make another attempt. 
So the timing for my system with the new changes is:
connect (and fail = 1 second) + 5 second delay + connect (with one second delay) + setup the devices (26 seconds) = 33 seconds.

I think this is a reasonable solution as it will only take this additional time during the setup of the integration and for most panels it will succeed to connect on the first attempt, saving additional 5 second delays.

Let me know what you guys think of it. My changes of yesterday still need some cleanup so it is not review ready yet, just for your information on what I did.

https://github.com/FredericMa/core/compare/add_communication_delay...FredericMa:core:add_communication_delay_v2?expand=1

Reply:
@OnFreund Can you share your thoughts (when you find the time) on my reply above and the related changes you can find in the compare at the bottom? Thanks.

Reply:
I guess this branch is now ready to merge.